src/core/qgsmaplayer.h

```diff
@@ -939,6 +939,11 @@ class CORE_EXPORT QgsMapLayer : public QObject
      */
     void clone( QgsMapLayer *layer ) const;
 
+    /** Metadata object for that layer.
+     * \since QGIS 3.0
+     */
+    QgsLayerMetadata mMetadata;
```

Comment:
This should be private to make sure it's always used via a setter, so a signal can be emitted when this changes in the future.

Reply:
Do I need to rollback?
I needed it to use it in child classes QgsRasterLayer and QgsVectorLayer.

Reply:
Use a getter for this in the child classes. (i.e. call `metadata()`), it's C++, not python ;)